core/startos/src/util/logger.rs

```diff
@@ -1,13 +1,62 @@
-use std::io;
+use std::fs::File;
+use std::io::{self, Write};
+use std::sync::{Arc, Mutex, MutexGuard};
 
+use lazy_static::lazy_static;
 use tracing::Subscriber;
+use tracing_subscriber::fmt::MakeWriter;
 use tracing_subscriber::util::SubscriberInitExt;
 
+lazy_static! {
+    pub static ref LOGGER: StartOSLogger = StartOSLogger::init();
+}
+
 #[derive(Clone)]
-pub struct EmbassyLogger {}
+pub struct StartOSLogger {
+    logfile: LogFile,
+}
+
+#[derive(Clone, Default)]
+struct LogFile(Arc<Mutex<Option<File>>>);
```

Comment:
Why did you use this a chance to get to logging to a file. I don't see the downsides, better rotations usually and keeping

Reply:
I did this because a large portion of init logs are typically lost when the disk gets mounted every time. This allows me to create an init log file at /run/startos/init.log that shows the logs from the last init